File: LeetCode/101-200/117. 填充每个节点的下一个右侧节点指针 II.java
/*
// Definition for a Node.
class Node {
    public int val;
    public Node left;
    public Node right;
    public Node next;

    public Node() {}
    
    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, Node _left, Node _right, Node _next) {
        val = _val;
        left = _left;
        right = _right;
        next = _next;
    }
};
*/

class Solution {
    public Node connect(Node root) {
       if (root == null) return  null;
       Node cur = root;
       while (cur != null) {
           Node nextTier = new Node(0);
           Node prev = nextTier;
           while (cur != null) {
               if (cur.left != null) {
                   prev.next = cur.left;
                   prev = prev.next;
               }
               if (cur.right != null) {
                   prev.next = cur.right;
                   prev = prev.next;
               }
               cur = cur.next;
           }
           cur = nextTier.next;
       }
        return  root;
    }
}

File: LeetCode/101-200/117. 填充每个节点的下一个右侧节点指针 II.go
/**
 * Definition for a Node.
 * type Node struct {
 *     Val int
 *     Left *Node
 *     Right *Node
 *     Next *Node
 * }
 */

 func connect(root *Node) *Node {
	cur := root
	for cur != nil {
		node := &Node{Val: 0}
		prev := node
		for cur != nil {
			if cur.Left != nil {
				prev.Next = cur.Left
				prev = prev.Next
			}
			if cur.Right != nil {
				prev.Next = cur.Right
				prev = prev.Next
			}
			cur = cur.Next
		}
		cur = node.Next
	}
	return root

}